### DISCUSSION TITLE
Replace some preprocessor flags with namelist variables in iHAMOCC

### DISCUSSION THREAD

**Discussion by TomasTorsvik** [NorESMhub/BLOM#166] [Ideas]
I would like to replace some preprocessor flags with namelist variables for iHAMOCC. At the moment preprocessor flags are used to turn on/off some basic functionality (e.g. cfc, natDIC, carbon isotopes), that I think would be better handled by a namelist variable. This will make it possible to change the model behavior without recompiling and creating a new executable. To start with, I want to replace FB_BGC_OCE, CFC, natDIC and CISONEW. 

We have a plan to expand the nitrogen cycle representation in iHAMOCC, and perhaps we would like to include other tracers later on. I believe it will become increasingly difficult to maintain the code if we continue adding preprocessor flags, so I would like to set a new structure before adding more tracers.

**Comment by TomasTorsvik** [comment]
Discussed with Jörg today. This is perhaps not as easy to do as I had hoped. In the current setup, the total number of tracers should be a fixed value (parameter), and in this case the use of preprocessor flags is hard to avoid (unless all tracers are always activated).

**Comment by jmaerz** [comment]
I guess, with the new PRs by Mariana, we can close this discussion :-)

**Comment by JorgSchwinger** [comment]
I agree this finally has been solved. Hooray!
